gmso/formats/lammpsdata.py

```diff
@@ -199,3 +209,228 @@ def write_lammpsdata(topology, filename, atom_style='full'):
+            if connection_type + 's' in line.split():
+                n_connections = int(line.split()[0])
+            if connection_type.capitalize() + 's' in line.split():
+                break
+    connection_lines = open(filename, 'r').readlines()[i+2:i+n_connections+2]
+    # Determine number of sites to generate
+    if connection_type == 'bond':
+        n_sites = 2
+    elif connection_type == 'angle':
+        n_sites = 3
+    else:
+        n_sites = 4
+    for i, line in enumerate(connection_lines):
+        site_list = list()
+        for j in range(n_sites):
+            site = topology.sites[int(line.split()[j+2])-1]
+            site_list.append(site)
+        if connection_type == 'bond':
+            connection = Bond(
+                connection_members=site_list,
+                connection_type=connection_type_list[int(line.split()[1])-1],
+                    )
+        elif connection_type == 'angle':
+            connection = Angle(
+                connection_members=site_list,
+                connection_type=connection_type_list[int(line.split()[1])-1],
+                    )
+        topology.add_connection(connection)
+
+    return topology
+
+def _get_atoms(filename, topology, unit_style, type_list):
+    """
+    Function to parse the atom information in the LAMMPS data file
+    """
+    with open(filename, 'r') as lammps_file:
+        for i, line in enumerate(lammps_file):
+            if 'atoms' in line.split():
+                n_atoms = int(line.split()[0])
+            if 'Atoms' in line.split():
+                break
+    atom_lines = open(filename, 'r').readlines()[i+2:i+n_atoms+2]
+    for line in atom_lines:
+        atom = line.split()
+        atom_type = atom[2]
+        charge = u.unyt_quantity(float(atom[3]), get_units(unit_style)['charge'])
+        coord = u.angstrom * u.unyt_array([
+            float(atom[4]),
+            float(atom[5]),
+            float(atom[6])])
+        site = Site(
+            charge=charge,
+            position=coord,
+            atom_type=type_list[int(atom_type)-1]
+            )
+        element = element_by_mass(site.atom_type.mass.value)
+        site.name = element.name
+        site.element = element
+        topology.add_site(site)
+
+    topology.update_sites()
+
+    return topology
+
+def _get_box_coordinates(filename, unit_style, topology):
+    """
+    Function to parse box information
+    """
+    with open(filename, 'r') as lammps_file:
+        for line in lammps_file:
+            if 'xlo' in line.split():
+                break
+        x_line = line.split()
+        y_line = lammps_file.readline().split()
+        z_line = lammps_file.readline().split()
+
+        x =  float(x_line[1])-float(x_line[0])
+        y =  float(y_line[1])-float(y_line[0])
+        z =  float(z_line[1])-float(z_line[0])
+
+        # Box Information
+        lengths = u.unyt_array([x,y,z], get_units(unit_style)['distance'])
```

Comment:
How do we want to handle the [tilt factors?](https://lammps.sandia.gov/doc/read_data.html)


>These are the recognized header keywords. Header lines can come in any order. The value(s) are read from the beginning of the line. Thus the keyword atoms should be in a line like “1000 atoms”; the keyword ylo yhi should be in a line like “-10.0 10.0 ylo yhi”; the keyword xy xz yz should be in a line like “0.0 5.0 6.0 xy xz yz”. All these settings have a default value of 0, except the lo/hi box size defaults are -0.5 and 0.5. A line need only appear if the value is different than the default.

Currently this will make orthogonal boxes.

Reply:
Good point, I need to look into this further.

Reply:
Tried to add coverage for triclinic boxes but it definitely needs to be validated.